config/confignet/confignet.go

```diff
@@ -39,11 +44,15 @@ type TCPAddr struct {
 	// If the host is a literal IPv6 address it must be enclosed in square brackets, as in "[2001:db8::1]:80" or
 	// "[fe80::1%zone]:80". The zone specifies the scope of the literal IPv6 address as defined in RFC 4007.
 	Endpoint string `mapstructure:"endpoint"`
+
+	// DialerTimeout is the maximum amount of time a dial will wait for
+	// a connect to complete. The default is no timeout.
+	DialerTimeout time.Duration `mapstructure:"dialer_timeout"`
 }
```

Comment:
Should we consider having "Dialer" as a separate struct with only Timeout for the moment in case we add more options there?

So essentially is a matter of writing:
```
dialer:
  timeout: 5s
  foo: ...
```
vs
```
dialer_timeout: 5s
dialer_foo: ...
```

Reply:
I like this idea, especially if we want to keep our options open for settings like `KeepAlive` or `Dealine`, etc.

Reply:
@bogdandrutu updated

Reply:
Instead of making our own struct should we embedded `net.Dialer`?

Reply:
NVM, we need mapstructure